src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadPackId.cs

```diff
@@ -43,4 +47,14 @@ public WorkloadPackId(string id)
 
         public static bool operator !=(WorkloadPackId a, WorkloadPackId b) => !a.Equals(b);
     }
+
+#if USE_SYSTEM_TEXT_JSON
+    internal class PackIdJsonConverter : JsonConverter<WorkloadPackId>
+    {
+        public override WorkloadPackId Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options) =>
+            new WorkloadPackId(reader.GetString() ?? string.Empty);
```

Comment:
If preview 7 serialised as `"Id":{}`, will [Utf8JsonReader.GetString](https://docs.microsoft.com/dotnet/api/system.text.json.utf8jsonreader.getstring?view=net-5.0#System_Text_Json_Utf8JsonReader_GetString) throw InvalidOperationException?

Reply:
Yes, but it will just show up as a warning which is fine as a cross preview conflict.